src/Symfony/Component/HttpKernel/Attribute/MapDateTime.php

```diff
@@ -11,14 +11,18 @@
 
 namespace Symfony\Component\HttpKernel\Attribute;
 
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\DateTimeValueResolver;
+
 /**
  * Controller parameter tag to configure DateTime arguments.
  */
 #[\Attribute(\Attribute::TARGET_PARAMETER)]
-class MapDateTime
+class MapDateTime extends ValueResolver
 {
     public function __construct(
-        public readonly ?string $format = null
+        public readonly ?string $format = null,
+        bool $disabled = false,
```

Comment:
why do we need disable resolvers?
Also why do we need introduce inheritance and `name` attribute? We've already have `'name' => RequestValueResolver::class`, so it is possible to build map `attribute => resolver` inside compiler pass

Reply:
> why do we need disable resolvers?

This generalizes `MapEntity::$disabled`.

> why do we need introduce inheritance

Because configuring a resolver for an argument is a hint you want it to provide a value.

> why do we need [the] name property? We've already extended DI with `'name' => RequestValueResolver::class`, so it is possible to build map `attribute => resolver` inside compiler pass

The `name` property is used to target a value resolver, and two value resolvers could have the same class so we cannot use their FQCN as key.

Not sure I understand what you meant though :sweat_smile:

Reply:
> build map attribute => resolver inside compiler pass

The current way is more powerful to end users, since they can decide which resolver to map to.
But I'd be fine adding an extra `string $resolver = DateTimeValueResolver::class` to this attribute, to really give all the power (and same in the other patched attributes).

Reply:
> adding an extra string $resolver

PR updated